### PR TITLE
Auto detect nvidia support

### PR DESCRIPTION
**Pull request by j-rivero**
This should fix the problems on current optimus introduced in #386, since `nvidia-toolkit-container` does not install `ENV NVIDIA_VISIBLE_DEVICES` variables needed to expose gpus into the container.

Other improvements:
 * Remove the need for `NVIDIA_DOCKER2_NODE` in nodes configuration. Autodetected from packages installed.
 * Remove the `libglvnd` installation. Testing seems not to need it.

Testing:
 * Optimus [Build Status](https://build.osrfoundation.org/job/_test_ignition_rendering-ci-default-bionic-amd64/22/)
 * Drogon (one failing, flaky) [Build Status](https://build.osrfoundation.org/job/_test_ignition_rendering-ci-default-bionic-amd64/23/)
 * Maria (no gpu) [Build Status](https://build.osrfoundation.org/job/_test_ignition_rendering-ci-default-bionic-amd64/24/)
 * r2d2 [Build Status](https://build.osrfoundation.org/job/_test_ignition_rendering-ci-default-bionic-amd64/25/)

### CODE REVIEW

File: jenkins-scripts/docker/lib/docker_run.bash [hunk 1]
```diff
@@ -41,29 +41,22 @@ if $USE_GPU_DOCKER; then
...
+        export EXTRA_PARAMS_STR="${EXTRA_PARAMS_STR} --gpus all"
+      ;;
+      'nvidia-docker2')
+        export EXTRA_PARAMS_STR="${EXTRA_PARAMS_STR} --runtime=nvidia"
+      ;;
+      'nvidia-docker')
+        export docker_cmd="nvidia-docker"
+      ;;
+      *)
+        echo "No docker-nvidia support was detected but Nvidia car is detected"
```
Comment (sloretz) [inline jenkins-scripts/docker/lib/docker_run.bash:56]: nit `but Nvidia car` -> `but an Nvidia card`

File: jenkins-scripts/docker/lib/docker_run.bash [hunk 2]
```diff
@@ -41,29 +41,22 @@ if $USE_GPU_DOCKER; then
...
+        echo "Probably a problem in provision of the agent"
```
Comment (sloretz) [inline jenkins-scripts/docker/lib/docker_run.bash:57]: nit `provision` -> `the provisioning`

**Review by sloretz — APPROVED** [review]
Looks like it works; I didn't see any issues with the code other than a couple typo's that were already there.

File: jenkins-scripts/docker/lib/docker_run.bash (hunk 1 above)
Comment (j-rivero) [reply jenkins-scripts/docker/lib/docker_run.bash]: this one is funny

**Comment by j-rivero** [comment]
Going to merge and enable the label in the node again.

**Pull request merged**